### PR TITLE
Helm work

### PR DESCRIPTION
**Pull request by Sedward**
I've done some work on adding the pvc and splitting out the deployments to make it easier to troubleshoot yaml errors. We may or may not want to merge this PR but I thought I'd try it. I've done some testing of my own and it mostly works but AWS is causing me issues with the the access mode because the only option I have is ReadWriteOnce.

### CODE REVIEW

File: airflow/templates/airflow-scheduler.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    metadata:
+      labels:
+        app: airflow
+        tier: scheduler
+    spec:
+      restartPolicy: Always
+      containers:
+      - name: {{ .Values.airflow.prefix -}}scheduler
+        image: {{ .Values.airflow.image }}
+        volumeMounts:
```
Comment (gsemet) [inline airflow/templates/airflow-scheduler.yaml:18]: I think you need to conditionalize this with `.Values.airflow.persistence.enabled`

File: airflow/templates/airflow-worker.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    metadata:
+      labels:
+        app: airflow
+        tier: worker
+    spec:
+      restartPolicy: Always
+      containers:
+      - name: {{ .Values.airflow.prefix -}}worker
+        image: {{ .Values.airflow.image }}
+        volumeMounts:
```
Comment (gsemet) [inline airflow/templates/airflow-worker.yaml:18]: here as well

**Review by gsemet — COMMENTED** [review]
look quite good to me! Thanks!

I'll test it next week when I go back to work, and it is work, I'll update the main PR with your proposal !

File: airflow/templates/pvc.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{{- if and .Values.airflow.persistence.enabled (not .Values.airflow.persistence.existingClaim) }}
```
Comment (gsemet) [inline airflow/templates/pvc.yaml:1]: arg, that's how we do a `and` in helm templates? That sucks! Why don't they use jinja?

File: airflow/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "airflow.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
```
Comment (gsemet) [inline airflow/templates/_helpers.tpl:11]: ho, good to know

File: airflow/templates/pvc.yaml (hunk 3 above)
Comment (Sedward) [reply airflow/templates/pvc.yaml:1]: I know. Luckily I pinched this example from the mysql helm chart. The go templating language has been a little tricky at times

**Comment by gsemet** [comment]
Hello. So I've integrated your change on my PR branch `helm_chart`. I had to rebase it and modify it, so I think you need to checkout the latest version of this branch and check it if it still work.

**Pull request closed without merging**